### PR TITLE
fixfees on grin-wallet

### PR DESCRIPTION
**Pull request by tromp**
Complements the fixfees PR at https://github.com/mimblewimble/grin/pull/3481
Still needs some doc tests fixed, which requires some assistance from @yeastplume

### CODE REVIEW

File: util/Cargo.toml [hunk 1]
```diff
@@ -35,20 +35,20 @@ sha3 = "0.8"
...
+# grin_store = { git = "https://github.com/mimblewimble/grin", branch = "master" }
 
 # For local testing
-# grin_core = { path = "../../grin/core"}
-# grin_keychain = { path = "../../grin/keychain"}
-# grin_chain = { path = "../../grin/chain"}
-# grin_util = { path = "../../grin/util"}
-# grin_api = { path = "../../grin/api"}
-# grin_store = { path = "../../grin/store"}
+grin_core = { git = "https://github.com/tromp/grin", branch = "fixfees" }
```
Comment (jaspervdm) [inline util/Cargo.toml:46]: We need to revert this file once the fixfees PR on grin is merged. Same with the Cargo.lock file

**Review by jaspervdm — COMMENTED** [review]
This PR increases the fee of a tx, will this cause compatibility issues with older wallets? If an older wallet initiates a tx with a newer wallet, will the newer wallet deny it because the fee is too low?

Other than this issue (and the Cargo files) I think the PR looks good.

File: util/Cargo.toml (hunk 1 above)
Comment (tromp) [reply util/Cargo.toml:46]: what you point out is not just a wallet issue. nodes running new code will not relay txs paying old fees.

**Review by jaspervdm — APPROVED** [review]
Upon further investigation there is no compatibility issue between upgraded and non-upgraded wallets, since the receiving side does not actually check the fee.

👍 from me once we fix the Cargo files after merging https://github.com/mimblewimble/grin/pull/3481.

File: api/src/owner_rpc.rs [hunk 2]
```diff
@@ -256,7 +256,7 @@ pub trait OwnerRpc {
 			  "confirmation_ts": "2019-01-15T16:01:26Z",
 			  "confirmed": true,
 			  "creation_ts": "2019-01-15T16:01:26Z",
-			  "fee": null,
+			  "fee_fields": null,
```
Comment (quentinlesceller) [inline api/src/owner_rpc.rs:259]: Just so we know this is a breaking change with the current API.

**Review by quentinlesceller — COMMENTED** [review]
Question about the API change below.

File: config/src/types.rs [hunk 3]
```diff
@@ -75,6 +78,7 @@ impl Default for WalletConfig {
 			tls_certificate_key: None,
 			dark_background_color_scheme: Some(true),
 			keybase_notify_ttl: Some(1440),
+			accept_fee_base: None,
```
Comment (quentinlesceller) [inline config/src/types.rs:81]: Shouldn't it be?
```suggestion
			accept_fee_base: default_accept_fee_base(),
```

File: controller/src/display.rs [hunk 4]
```diff
@@ -412,17 +413,17 @@ pub fn estimate(
 		bMG->"Will be locked",
 	]);
 
-	for (strategy, total, fee) in strategies {
+	for (strategy, total, fee_fields) in strategies {
 		if dark_background_color_scheme {
 			table.add_row(row![
 				bFC->strategy,
-				FR->amount_to_hr_string(fee, false),
+				FR->amount_to_hr_string(fee_fields.fee(0), false),
```
Comment (quentinlesceller) [inline controller/src/display.rs:420]: Is it safe to use height 0 here?

File: libwallet/src/api_impl/owner.rs [hunk 5]
```diff
@@ -402,8 +402,8 @@ where
 	let amount = if tx.amount_credited >= tx.amount_debited {
 		tx.amount_credited - tx.amount_debited
 	} else {
-		let fee = match tx.fee {
-			Some(f) => f,
+		let fee = match tx.fee_fields {
+			Some(f) => f.fee(0),
```
Comment (quentinlesceller) [inline libwallet/src/api_impl/owner.rs:406]: Same as above and actually it's kinda weird to use height 0 to have the value no? Wouldn't introduce a new function would be cleaner?

**Review by j01tz — COMMENTED** [review]
:+1: once API conflict above is resolved. Will still require heavy testing in beta.

**Review by quentinlesceller — COMMENTED** [review]
Question about the display of tx_fees below. Other than that I didn't catch anything.

File: config/src/types.rs (hunk 3 above)
Comment (tromp) [reply config/src/types.rs:81]: if accept_fee_base is not configured then it falls back to the global default. so these would seem to be equivalent.

File: api/src/owner_rpc.rs (hunk 2 above)
Comment (jaspervdm) [reply api/src/owner_rpc.rs:259]: This also breaks compatibility with older wallets that already have tx log entries stored in their wallet db. This is not intended, will fix.

File: controller/src/display.rs (hunk 4 above)
Comment (tromp) [reply controller/src/display.rs:420]: I guess hardfork4 height is better here.

**Review by j01tz — APPROVED** [review]

**Pull request merged**